spec/features/profile_spec.rb

```diff
@@ -1,23 +1,32 @@
 require 'rails_helper'
 
-feature 'Edit profile' do
-  let(:user) { create(:user) }
+RSpec.feature 'Edit profile' do
   let(:features) { FeaturesSpecHelper.new }
 
-  scenario 'A user edits its profile information and password' do
+  background do
+    @old_password = user.encrypted_password
     features.sign_in(user)
 
     click_on 'See my profile'
 
     click_on 'Edit profile'
 
-    select 'Sergent', from: 'user_rank'
+    select 'Caporal', from: 'user_rank'
     fill_in 'user_password', with: 'new_password'
     fill_in 'user_password_confirmation', with: 'new_password'
     fill_in 'user_current_password', with: user.password
     click_button 'Update'
 
-    expect(page).to have_content 'Your account has been updated successfully.'
-    expect(User.find_by(email: user.email).rank).to eq('sgt')
+    @updated_user = User.find_by(email: user.email)
```

Comment:
We should only test the UI in feature test.
It there a reason why you don't test the flash message here?